core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -95,16 +118,32 @@ public static MetricsConfig fromProperties(Map<String, String> props) {
           try {
             mode = MetricsModes.fromString(props.get(key));
           } catch (IllegalArgumentException err) {
-            // Mode was invalid, log the error and use the default
+            // Mode was invalid, log the error and use the default (or default for sorted columns)
             LOG.warn("Ignoring invalid metrics mode for column {}: {}", columnAlias, props.get(key), err);
-            mode = spec.defaultMode;
+            if (sortedCols.contains(columnAlias)) {
```

Comment:
If you remove the if condition above, you will be able to avoid this change too.
I think something as below would work.

```
if (order != null) {
  MetricsMode sortColumnDefaultMode = sortColumnDefaultMode(defaultMode);
  Set<String> sortColumns = SortOrderUtil.sortColumns(order);
  for (String sortColumn : sortColumns) {
    spec.columnModes.put(sortColumn, sortColumnDefaultMode);
  }
}
```

Reply:
The purpose of this if,  if it's an invalid metric for a sorted column, we pick the sorted default instead of the normal default.

It's probably over-engineering though?  I reverted this and edited the test that was asserting this